packages/interface/src/lib.ts

```diff
@@ -499,6 +504,10 @@ export interface PrincipalParser {
   parse(did: UCAN.DID): Verifier
 }
 
+export interface DIDResolver {
+  resolveDID?: (did: UCAN.DID) => Await<Result<DIDKey, DIDResolutionError>>
```

Comment:
For the long term, I bristle a bit seeing the only thing that a `UCAN.DID` can resolve to is a single `DIDKey`.

What if we had 
```
resolveDID<DidMethod>?: (did: UCAN.DID<DidMethod>) => Await<Result<DIDResolution<DidMethod>, DIDResolutionError>>
```

and

```
type SingleKeypairDidDocument<ID extends UCAN.DID> = {
  id: ID,
  /** https://www.w3.org/TR/did-core/#also-known-as */
  alsoKnownAs: DIDKey
}
type DidResolution<ID> =
| KeyAliasDidDocument<ID>
```

Reply:
alternatively could have `resolveDID` -> `resolveDIDKey`.

I like that it gets nowhere near 'did resolution' nor reinforces the (IMO footgun) assumption that a DID will always cleanly resolve to a single default keypair.

Reply:
some more on did resolution result https://w3c-ccg.github.io/did-resolution/#did-resolution-result

Reply:
and impl https://github.com/decentralized-identity/did-resolver/blob/master/src/resolver.ts#L27

Reply:
You are right `resolveDID` is misleading name, this is not a function to resolve a DID document but rather to resolve a key. I'll rename both interface and a function.